azkaban-common/src/main/java/azkaban/project/FlowTrigger.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.project;
+
+import com.google.common.base.Preconditions;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * FlowTrigger is the logical representation of a trigger.
+ * It couldn't be changed once gets constructed.
+ * It will be used to create running trigger instance.
+ */
+public class FlowTrigger {
+
+  private final List<FlowTriggerDependency> dependencies;
+  private final CronSchedule schedule;
+  private final Duration maxWaitDuration;
+
+  /**
+   * @throws IllegalArgumentException if any of the argument is null
+   * @throws IllegalArgumentException if there is duplicate dependency name
+   * @throws IllegalArgumentException if there is duplicate dependency type and params
```

Comment:
Change to one @throws with longe description?

Reply:
updated